redisvl/llmcache/semantic.py

```diff
@@ -180,14 +184,18 @@ def check(
             return_score=True,
         )
 
-        results = self._index.search(v.query, query_params=v.params)
+        cache_hits = self._index.query(v, process_results=self._process_cache_hits)
+        return cache_hits
 
-        cache_hits = []
+    def _process_cache_hits(self, results: List["Result"]) -> List[str]:
+        cache_hits: List[str] = []
         for doc in results.docs:
             sim = similarity(doc.vector_distance)
             if sim > self.threshold:
                 self._refresh_ttl(doc.id)
-                cache_hits.append(doc.response)
+                cache_hits.append(
+                    doc.response
+                )  # TODO what do we actually want to return here?
```

Comment:
Allow the user to specify fields to return in ``check``? defaults to ``response`` only. 

maybe another method like ``check_and_get`` that returns a list of dicts?